ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala

```diff
@@ -169,15 +175,39 @@ object LedgerContext {
 
     require(esf != null, "ExecutionSequencerFactory must not be null.")
 
-    def ledgerId: String =
+    var cachedLedgerId: Option[domain.LedgerId] = None
+
+    def ledgerId: domain.LedgerId =
       configuredLedgerId match {
-        case LedgerIdMode.Static(id) => id.unwrap
+        case LedgerIdMode.Static(id) => id
         case LedgerIdMode.Dynamic() =>
-          LedgerIdentityServiceGrpc
-            .blockingStub(channel)
-            .getLedgerIdentity(GetLedgerIdentityRequest())
-            .ledgerId
+          cachedLedgerId.synchronized {
+            cachedLedgerId match {
+              case Some(id) =>
+                id
+              case None =>
+                val id =
+                  domain.LedgerId(
+                  LedgerIdentityServiceGrpc
+                    .blockingStub(channel)
```

Comment:
making a blocking call while holding a lock is not a good idea in general. I'd rather have an async behaviour and cache the `Promise` instead stored in an `AtomicRefence`